src/auto_code_surveys.py

```diff
@@ -3,26 +3,41 @@
 
 from core_data_modules.cleaners import Codes
 from core_data_modules.cleaners.cleaning_utils import CleaningUtils
+from core_data_modules.logging import Logger
 from core_data_modules.traced_data import Metadata
 from core_data_modules.traced_data.io import TracedDataCodaV2IO
 from core_data_modules.util import IOUtils
+from dateutil.parser import isoparse
 
 from src.lib.code_schemes import CodeSchemes
 from src.lib.pipeline_configuration import PipelineConfiguration
 
+log = Logger(__name__)
+
 
 class AutoCodeSurveys(object):
     SENT_ON_KEY = "sent_on"
 
     @classmethod
-    def auto_code_surveys(cls, user, data, coda_output_dir):
+    def auto_code_surveys(cls, user, data, pipeline_configuration, coda_output_dir):
         # Auto-code surveys
         for plan in PipelineConfiguration.SURVEY_CODING_PLANS:
             for cc in plan.coding_configurations:
                 if cc.cleaner is not None:
                     CleaningUtils.apply_cleaner_to_traced_data_iterable(user, data, plan.raw_field, cc.coded_field,
                                                                         cc.cleaner, cc.code_scheme)
 
+        # Remove survey data sent after the project finished
+        log.info("Removing survey messages sent after the end of the project")
+        out_of_range_count = 0
+        for td in data:
+            for plan in PipelineConfiguration.SURVEY_CODING_PLANS:
+                if plan.time_field in td and isoparse(td[plan.time_field]) > pipeline_configuration.project_end_date:
+                    out_of_range_count += 1
+                    td.hide_keys({plan.raw_field, plan.time_field},
+                                 Metadata(user, Metadata.get_call_location(), time.time()))
+        log.info(f"Removed {out_of_range_count} survey messages sent after the end of the project")
```

Comment:
Consider 'hidden' rather than 'removed' given that this will still leave them in the tracedata, just not 'publicly' visible.

Reply:
Changed (and added a note to the comment explaining what the hiding action does)